src/components/encounter/ohri-encounter-form.component.tsx

```diff
@@ -260,9 +260,12 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
           // for each question and answer, see if we find a matching concept, and, if so:
           //   1) replace the concept reference with uuid (for the case when the form references the concept by mapping)
           //   2) use the concept display as the label if no label specified
-          const matchingConcept = findConceptByReference(field.questionOptions.concept, concepts);
-          field.questionOptions.concept = matchingConcept ? matchingConcept.uuid : field.questionOptions.concept;
-          field.label = field.label ? field.label : matchingConcept?.display;
+          const matchingConcept = findConceptByReference(field.questionOptions.concept, concepts); 
+          field.meta = matchingConcept;
+          field.questionOptions.concept = field.meta ? field.meta?.uuid : field.questionOptions.concept;
+         
+
+          field.label = field.label ? field.label : field.meta?.display;
```

Comment:
```suggestion
          const matchingConcept = findConceptByReference(field.questionOptions.concept, concepts);
          field.questionOptions.concept = matchingConcept ? matchingConcept.uuid : field.questionOptions.concept;
          field.label = field.label ? field.label : matchingConcept?.display;
          field.meta.concept = matchingConcept;
```

Take note of the concept path.